### PR TITLE
chore: fix typescript benchmarks by removing ts-benchmark

### LINKED ISSUE #12770: TS benchmark fails with TypeScript 4.9
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

See https://github.com/Automattic/mongoose/pull/12769 . TS benchmark fails with `Current⠀benchmark⠀result⠀268025⠀>⠀100000` locally when using TS 4.9. Looks like 4.9 introduced some serious performance regression :sob:

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix #12770

**Summary**

I haven't been able to figure out why the ts-benchmark module is failing. The fact that they ship minified JS to npm makes it tricky to step through the code, but I figured it isn't especially difficult to write a quick script that parses `tsc --extendedDiagnostics` output. One less dependency to worry about too.

**Examples**

### CODE REVIEW

File: scripts/tsc-diagnostics-check.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const fs = require('fs');
+
+const stdin = fs.readFileSync(0).toString('utf8');
+const maxInstantiations = isNaN(process.argv[2]) ? 100000 : +process.argv[2];
```
Comment (hasezoey) [inline scripts/tsc-diagnostics-check.js:6]: why use `+process.argv[2]` and not `parseInt()`?

File: scripts/tsc-diagnostics-check.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+console.log(stdin);
+
+const numInstantiations = stdin.match(/Instantiations:\s+(\d+)/)[1];
```
Comment (hasezoey) [inline scripts/tsc-diagnostics-check.js:10]: maybe consider also using `parseInt()` here

**Review by hasezoey — COMMENTED** [review]
generally looks good to me, though i have some questions regarding consistency

**Review by hasezoey — APPROVED** [review]
LGTM

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**